internal/webrtc/tuya/mqtt.go

```diff
@@ -0,0 +1,274 @@
+
+	log.Print("subscribe mqtt topic success")
+}
+
+func (t *tuyaSession) StopMqtt() {
+	t.mqtt.client.Disconnect(1000)
+}
+
+func (t *tuyaSession) sendOffer(sessionID string, sdp string) {
+	offerFrame := struct {
+		Mode       string `json:"mode"`
+		Sdp        string `json:"sdp"`
+		StreamType uint32 `json:"stream_type"`
+		Auth       string `json:"auth"`
+	}{
+		Mode:       "webrtc",
+		Sdp:        sdp,
+		StreamType: 1,
+		Auth:       t.mqtt.auth,
+	}
+
+	offerMqtt := &MqttMessage{
+		Protocol: 302,
+		Pv:       "2.2",
+		T:        time.Now().Unix(),
+		Data: MqttFrame{
+			Header: MqttFrameHeader{
+				Type:      "offer",
+				From:      t.mqtt.MQTTUID,
+				To:        t.config.DeviceID,
+				SubDevID:  "",
+				SessionID: sessionID,
+				MotoID:    t.mqtt.motoID,
+			},
+			Message: offerFrame,
+		},
+	}
+
+	sendBytes, err := json.Marshal(offerMqtt)
+	if err != nil {
+		log.Printf("marshal offer mqtt to bytes fail: %s", err.Error())
+
+		return
+	}
+
+	t.mqttPublish(sendBytes)
+}
+
+func (t *tuyaSession) sendCandidate(sessionID string, candidate string) {
+	candidateFrame := struct {
+		Mode      string `json:"mode"`
+		Candidate string `json:"candidate"`
+	}{
+		Mode:      "webrtc",
+		Candidate: candidate,
+	}
+
+	candidateMqtt := &MqttMessage{
+		Protocol: 302,
+		Pv:       "2.2",
+		T:        time.Now().Unix(),
+		Data: MqttFrame{
+			Header: MqttFrameHeader{
+				Type:      "candidate",
+				From:      t.mqtt.MQTTUID,
+				To:        t.config.DeviceID,
+				SubDevID:  "",
+				SessionID: sessionID,
+				MotoID:    t.mqtt.motoID,
+			},
+			Message: candidateFrame,
+		},
+	}
+
+	sendBytes, err := json.Marshal(candidateMqtt)
+	if err != nil {
+		log.Printf("marshal candidate mqtt to bytes fail: %s", err.Error())
+
+		return
+	}
+
+	t.mqttPublish(sendBytes)
+}
+
+// 发布mqtt消息
+func (t *tuyaSession) mqttPublish(payload []byte) {
+	token := t.mqtt.client.Publish(t.mqtt.publishTopic, 1, false, payload)
+	if token.Error() != nil {
+		log.Printf("mqtt publish fail: %s, topic: %s", token.Error().Error(),
+			t.mqtt.publishTopic)
+	}
+}
+
+func (t *tuyaSession) mqttConsume(msg mqtt.Message) {
+	tmp := struct {
+		Protocol int    `json:"protocol"`
+		Pv       string `json:"pv"`
+		T        int64  `json:"t"`
+		Data     struct {
+			Header  MqttFrameHeader `json:"header"`
+			Message json.RawMessage `json:"msg"`
+		} `json:"data"`
+	}{}
+
+	if err := json.Unmarshal(msg.Payload(), &tmp); err != nil {
+		log.Printf("unmarshal received mqtt fail: %s, payload: %s", err.Error(), string(msg.Payload()))
+
+		return
+	}
+
+	rmqtt := &MqttMessage{
+		Protocol: tmp.Protocol,
+		Pv:       tmp.Pv,
+		T:        tmp.T,
+		Data: MqttFrame{
+			Header:  tmp.Data.Header,
+			Message: tmp.Data.Message,
+		},
+	}
+
+	log.Printf("mqtt recv message, session: %s, type: %s, from: %s, to: %s",
+		rmqtt.Data.Header.SessionID,
+		rmqtt.Data.Header.Type,
+		rmqtt.Data.Header.From,
+		rmqtt.Data.Header.To)
+
+	t.mqttDispatch(rmqtt)
+}
+
+// 分发从mqtt服务器接受到的消息
```

Comment:
I think you should cleanup these comments in Chinese. Or at least translate them, if they are important.